vault/identity_store_oidc_provider.go

```diff
@@ -200,6 +363,19 @@ func (i *IdentityStore) pathOIDCReadAssignment(ctx context.Context, req *logical
 func (i *IdentityStore) pathOIDCDeleteAssignment(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
 	name := d.Get("name").(string)
 	err := req.Storage.Delete(ctx, assignmentPath+name)
+
```

Comment:
I think we want to delete the assignment after the client reference checks? Currently, the assignment is deleted first and then the checks take place.